code/modules/wiremod/components/list/list_pick.dm

```diff
@@ -54,24 +54,24 @@
 
 
 /obj/item/circuit_component/list_pick/input_received(datum/port/input/port)
-	if(parent.Adjacent(user.value))
-		return
-
-	if(ismob(user.value))
+	if(isliving(user.value))
+		var/mob/living/living_user = user.value
+		if(!(living_user.can_perform_action(parent.shell, FORBID_TELEKINESIS_REACH, ALLOW_SILICON_REACH, ALLOW_RESTING)))
+			return
 		trigger_output.set_output(COMPONENT_SIGNAL)
 		INVOKE_ASYNC(src, PROC_REF(show_list), user.value, input_name.value, input_list.value)
 
 /// Show a list of options to the user using standed TGUI input list
-/obj/item/circuit_component/list_pick/proc/show_list(mob/user, message, list/showed_list)
+/obj/item/circuit_component/list_pick/proc/show_list(mob/living/u, message, list/showed_list)
```

Comment:
I'm not sure why you're changing user to a single letter variable name but it removes information, so please revert it.

Reply:
Was more to imply/ensure it was from LIVING and not from MOB. also I have a circuit input named  user so name conflict happend which meant it was not clear which user he used

Reply:
I made the name a bit clearer, but making it user would be not great due to the mentioned conflict issue

Reply:
could name it something more detailed like `showing_to`